### DISCUSSION TITLE
ModalProvider unmounts previous modals

### DISCUSSION THREAD

**Discussion by testerez** [mantinedev/mantine#4023] [Feature requests]
### What package has an issue

@mantine/modals

### Describe the bug

I noticed that `ModalProvider` only keeps the currently visible modal mounted. This can be an issue when you open a modal from another one. For example if you display a form in a modal and want to open a confirmation modal on top of it. In that case, the state of the form is lost. See my sandbox link for a self contained example.

### What version of @mantine/hooks page do you have in package.json?

6.0.6

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/agitated-microservice-bt1k2f?file=/src/App.tsx

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

It would be nice if `ModalProvider` could keep all opened modals mounted so that we can stack them without loosing the components state.

**Comment by testerez** [comment]
Hey @rtivital , is there any chance to see this change implemented? Otherwise I may code my own modal provider so that I can stack modals without loosing state.

**Reply by andrew-stockdale-paytently** [reply]
Sorry this is an old discussion but I'd also really like to know if there are any plans for this. The Modal Manager not working with the keepMounted prop makes a lot of more complex modal flows a massive headache to implement

**Reply by rtivital** [reply]
Currently there are no plans to change anything in the modals manager – it was not designed to handle complex cases. If you want to do something that is not supported by `@mantine/modals` package, use `Modal` component from `@mantine/core` package instead.
